src/foam/nanos/http/AuthWebAgent.java

```diff
@@ -169,7 +171,20 @@ public Session authenticate(X x) {
           .put(HttpServletRequest.class,  req)
           .put(HttpServletResponse.class, resp), email, password);
 
-        if ( user != null ) return session;
+        if ( user != null ) {
+          if ( ! SafetyUtil.isEmpty(actAs) ) {
+            AgentAuthService agentService = (AgentAuthService) x.get("agentAuth");
+            DAO localUserDAO = (DAO) x.get("localUserDAO");
+            try {
+              User entity = (User) localUserDAO.find(Long.parseLong(actAs));
+              agentService.actAs(session.getContext(), entity);
+            } catch (java.lang.NumberFormatException e) {
+              logger.error("actAs must be a number:" + e);
```

Comment:
Can't we support email as well, since that's what user use to login, not their userId.

Reply:
That should work, but it would have to be the email of the business, and in order to look that up they would (I think) need the id anyway. Also, there may be a business and a user with the same email ID, which might be confusing.

Otherwise the way it works in the gui is they are shown a list of businesses they can actAs when they sign in, and pick one. We should have a call for them to get a similar list of businesses, but after that I think id as the identifier still makes sense to me.

Reply:
> Can't we support email as well, since that's what user use to login, not their userId.

@kgrgreer Correct, but you don't use an email to choose which business to act as.